voctopublish/api_client/googlechat_client.py

```diff
@@ -0,0 +1,113 @@
+from pathlib import Path
+
+from requests import post
+
+
+def send_chat_message(ticket, config):
+    logging.info("posting message to google chat")
+
+    message = {
+        "cards": [
+            {
+                "header": {
+                    "title": ticket.title,
+                    "subtitle": ticket.acronym,
+                },
+                "sections": [
+                    {
+                        "widgets": [
+                            {
+                                "textParagraph": {
+                                    "text": ticket.abstract,
+                                },
+                            },
+                        ],
+                    },
+                    {
+                        "widgets": [
+                            {
+                                "keyValue": {
+                                    "topLabel": "Speakers",
+                                    "content": ", ".join(ticket.people),
+                                },
+                            },
+                            {
+                                "keyValue": {
+                                    "topLabel": "Track",
+                                    "content": ticket.track if ticket.track else "-",
+                                },
+                            },
+                        ],
+                    },
+                    {
+                        "widgets": [
+                            {
+                                "buttons": [],
+                            },
+                        ],
+                    },
+                ],
+            },
+        ],
+    }
+
+    if ticket.voctoweb_enable and ticket.profile_voctoweb_enable:
+        message["cards"][0]["sections"][2]["widgets"][0]["buttons"].append(
+            {
+                "textButton": {
+                    "text": config["voctoweb"]["instance_name"],
+                    "onClick": {
+                        "openLink": {
+                            "url": config["voctoweb"]["frontend_url"]
+                            + "/v/"
+                            + ticket.slug
+                        },
+                    },
+                }
+            }
+        )
+
+    if ticket.youtube_enable and ticket.profile_youtube_enable:
+        message["cards"][0]["sections"][2]["widgets"][0]["buttons"].append(
+            {
+                "textButton": {
+                    "text": "YouTube",
+                    "onClick": {
+                        "openLink": {
+                            "url": ticket.youtube_urls["YouTube.Url0"],
+                        },
+                    },
+                }
+            }
+        )
+
+    if not message["cards"][0]["sections"][2]["widgets"][0]["buttons"]:
+        logging.warning("GoogleChat: No buttons for videos :(")
+        return
+
+    try:
+        logging.debug(repr(message))
+        r = post(ticket.googlechat_webhook_url, json=message)
```

Comment:
Somehow, this currently gets me an empty message

Reply:
Google Chat is stupid, if you have empty `keyValue` fields, it just ignores the whole message and outputs nothing at all.